src/Sulu/Component/Rest/ListBuilder/Metadata/AbstractPropertyMetadata.php

```diff
@@ -18,195 +20,120 @@
  */
 abstract class AbstractPropertyMetadata
 {
-    /**
-     * @var string
-     */
-    private $name;
-
-    /**
-     * @var string
-     */
-    private $translation;
-
-    /**
-     * @var string
-     */
-    private $visibility = FieldDescriptorInterface::VISIBILITY_NEVER;
-
-    /**
-     * @var string
-     */
-    private $searchability = FieldDescriptorInterface::SEARCHABILITY_NEVER;
-
-    /**
-     * @var string
-     */
-    private $type = 'string';
-
-    /**
-     * @var bool
-     */
-    private $sortable = true;
-
-    /**
-     * @var array
-     */
-    private $transformerTypeParameters;
-
-    /**
-     * @var string
-     */
-    private $filterType;
-
-    /**
-     * @var array
-     */
-    private $filterTypeParameters;
-
-    /**
-     * @var string
-     */
-    private $width;
-
-    public function __construct($name)
+    private string $name;
+
+    private string $translation;
+
+    private string $visibility = FieldDescriptorInterface::VISIBILITY_NEVER;
+
+    private string $searchability = FieldDescriptorInterface::SEARCHABILITY_NEVER;
+
+    private string $type = 'string';
+
+    private bool $sortable = true;
+
+    /** @var array<mixed> */
+    private array $transformerTypeParameters;
+
+    private string $filterType;
+
+    /** @var array<mixed> */
+    private array $filterTypeParameters;
+
+    private string $width;
+
+    public function __construct(string $name)
     {
         $this->name = $name;
         // default for translation can be overwritten by setter
         $this->translation = \ucfirst($name);
     }
 
-    /**
-     * @return string
-     */
-    public function getName()
+    public function getName(): string
     {
         return $this->name;
     }
 
-    /**
-     * @return string
-     */
-    public function getTranslation()
+    public function getTranslation(): string
     {
         return $this->translation;
     }
 
-    /**
-     * @param string $translation
-     */
-    public function setTranslation($translation)
+    public function setTranslation(string $translation): void
```

Comment:
The return and method parameter types need to be moved to phpdoc so we are not doing here any unneeded bc breaks.